testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedRegistrationTest.java

```diff
@@ -0,0 +1,54 @@
+package org.keycloak.testsuite.actions;
+
+import jakarta.mail.MessagingException;
+import org.jboss.arquillian.graphene.page.Page;
+import org.junit.Before;
+import org.junit.Test;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.testsuite.AbstractTestRealmKeycloakTest;
+import org.keycloak.testsuite.Assert;
+import org.keycloak.testsuite.admin.ApiUtil;
+import org.keycloak.testsuite.pages.AppPage;
+import org.keycloak.testsuite.pages.RegisterPage;
+
+import java.io.IOException;
+
+public class AppInitiatedRegistrationTest extends AbstractTestRealmKeycloakTest {
+
+    @Page
+    protected AppPage appPage;
+
+    @Page
+    protected RegisterPage registerPage;
+
+    @Override
+    public void configureTestRealm(RealmRepresentation testRealm) {
+    }
+
+    @Before
+    public void before() {
+        ApiUtil.removeUserByUsername(testRealm(), "test-user@localhost");
+    }
+
+    @Test
+    public void ensureLocaleParameterIsPropagatedDuringAppInitiatedRegistration() throws IOException, MessagingException {
+
+        String appInitiatedRegisterUrl = oauth.getLoginFormUrl();
+        appInitiatedRegisterUrl = appInitiatedRegisterUrl.replace("openid-connect/auth", "openid-connect/registrations");
+
+        // add the kc_locale parameter
+        appInitiatedRegisterUrl += "&kc_locale=en";
```

Comment:
I'd prefer we add query params with some kind of utility class like `UriBuilder`, as it is less error prone.